### PR TITLE
Pricing page rework: Add featured product cards

### PR DESCRIPTION
**Pull request by manzoorwanijk**
#### Proposed Changes

* Add assets to `client/assets/images/jetpack`
* Add `HeroImage` component
* Create reusable `FeaturedItemCard` component to use for both products and bundles
* Create `useCreateCheckout` hooks to keep components clean
* Enhance `DisplayPrice` component to allow a change in the order of original and discounted price
* Add `FeaturedItemCard` both in product and bundle list

#### Testing Instructions

* click on Jetpack Cloud live link below
    * Goto `/pricing?flags=jetpack/pricing-page-rework-v1`
* or boot up this PR 
    * Run `git fetch && git checkout add/jp-pricing-rework/featured-product-cards`
    * Run `yarn start-jetpack-cloud`
    * Goto [http://jetpack.cloud.localhost:3000/pricing?flags=jetpack/pricing-page-rework-v1](http://jetpack.cloud.localhost:3000/pricing?flags=jetpack/pricing-page-rework-v1)
* Confirm that you see "Backup" and "VideoPress" featured cards in "Most popular products"
* Goto Bundles view
* Confirm that you see "Security" and "Complete" featured cards under "Most popular bundles"
* Confirm that the design matches what is in Firma

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Completes: 1202796695664022-as-1202796695664102/f and 1202796695664022-as-1202796695664123/f

Screenshot 2022-08-29 at 5 38 46 PM
Screenshot 2022-08-29 at 5 38 55 PM
Screenshot 2022-08-29 at 5 39 43 PM
Screenshot 2022-08-29 at 5 39 54 PM

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202796695664123
  - 0-as-1202796695664102

### CODE REVIEW

File: client/my-sites/plans/jetpack-plans/product-store/hooks/use-create-checkout.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	);
+
+	return useMemo(
+		() => ( {
+			getCheckoutURL,
+			getOnClickPurchase,
+		} ),
+		[ getCheckoutURL, getOnClickPurchase ]
+	);
+};
```
Comment (manzoorwanijk) [inline client/my-sites/plans/jetpack-plans/product-store/hooks/use-create-checkout.ts]: This logic is extracted from `client/my-sites/plans/jetpack-plans/product-card/index.tsx`

**Comment by elliottprogrammer** [comment]
A couple things I noticed, although possibly not related to this PR, but worth looking into in followup PR's:
1. Currently the ProductStore page content max-width (on desktop screens) is set at 1040px.  Looking at the Figma designs, it appears that the max desktop width should be set to 1192px. I created a new Asana task for it here: 1202796695664022-as-1202886227656193/f
2. For some reason when viewing the changes in Jetpack Cloud Live build, there appears to be some conflicting/missing css styles. The product cards CTA buttons have an incorrect border styles (black solid borders & no border radius).  Also the page headings have lost their styles.  Viewing the changes locally, everything looks good!  Not sure what's happening here but maybe worth looking into to make sure these issues don't persist into production. 🤷‍♂️  (See Screenshot comparisons below)


local build (jetpack.cloud.localhost:3000)          |  Jetpack Cloud Live environment
:----------------------------------------------:|:----------------------------------:
  |

**Review by elliottprogrammer — APPROVED** [review]
This all looks very good to me! All the images appear to respond correctly (and in correct orientation) in all viewport sizes. Nice work! The code looks all good and very clean and readable. 🙂
LGTM! 👍

**Comment by manzoorwanijk** [comment]
> 2\. For some reason when viewing the changes in Jetpack Cloud Live build, there appears to be some conflicting/missing css styles. The product cards CTA buttons have an incorrect border styles (black solid borders & no border radius).  Also the page headings have lost their styles.

@elliottprogrammer here is a screenshot of what I see on Jetpack Cloud Live link

Screenshot 2022-08-30 at 12 00 17 PM

**Comment by manzoorwanijk** [comment]
> 1. Currently the ProductStore page content max-width (on desktop screens) is set at 1040px.  Looking at the Figma designs, it appears that the max desktop width should be set to 1192px. I created a new Asana task for it here

I have fixed it in this PR as suggested by Ilona.

**Comment by manzoorwanijk** [comment]
Slack thread with @ilonagl for the fixes applied: p1661775226634749-slack-C03RCHNVC0H

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7504587

Thank you @manzoorwanijk for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7504587).

**Pull request merged**